src/index.ts

```diff
@@ -239,6 +239,46 @@ if(!eos.PublicKey.isValid(data)) {
   return eos.PublicKey(data).toBuffer();
 }
 
+function tezosAddressEncoder(data: Buffer): string {
```

Comment:
Can you reuse the `makeBase58CheckEncoder` and `makeBase58CheckDecoder` functions? These functions seem to duplicate their functionality.

Reply:
Tezos addresses are not p2pkh addresses and so cannot use that decoder.

Reply:
They don't need to be - the base58check encoder/decoder supports aribtrary prefixes. It should be compatible with this format; if it's not, it should be easy to make it compatible. The current code largely replicates that function.

Reply:
I think `makeBase58CheckEncoder`/`makeBase58CheckDecoder` names are just confusing as the majority of the code were checking the optcode sequence of p2pkh/p2h. The only things they share are that they call bs58check.encode/decode (which are now called `bs58Decode/bs58Encode` ).  If these names are confusing, `makeBase58Check*` should just be renamed into `makeBitcoinBase58Check*`.

Reply:
@makoto They're not specific to p2pkh - they should work with any prefix.